ipatests/test_integration/test_installation.py

```diff
@@ -1836,6 +1836,43 @@ def test_replica_install_against_server(self):
                                            stdin_text=dirman_password)
         assert self.replicas[0].hostname not in cmd.stdout_text
 
+    def test_replica_install_existing_agreement(self):
+        """When the replication agreement already exists, the installer
+        should print an error message with the information on where
+        and how to remove it"""
+        # master and replica_1 already installed by class' install() method
+        tasks.stop_ipa_server(self.master)
+
+        # intentionally using ipa-server-install --unistall
+        # instead of replica-manage
+        self.replicas[0].run_command(['ipa-server-install',
+                                      '--uninstall',
+                                      '--force',
+                                      '-U'])
+        tasks.start_ipa_server(self.master)
+
+        result = tasks.install_replica(self.master,
+                                       self.replicas[0],
+                                       setup_ca=False,
+                                       setup_dns=True,
+                                       promote=False)
```

Comment:
Please set raiseonerr=False if you are expecting it to fail.

Reply:
Thank you, Anuja! Fixed.